src/components/DatasetsDropdown.js

```diff
@@ -16,13 +16,17 @@ export const DatasetsDropdown = props => (
             onChange={props.onChange ? props.onChange : props.selectDataSet}
             menuItems={props.options}
             emptyLabel={i18n.t('Select Data Set')}
-            hintText={i18n.t('Select Data Set')}
+            hintText={i18n.t(
+                props.loading ? 'Loading options...' : 'Select Data Set'
+            )}
```

Comment:
I don't think this will work. You need to do something like this:
```javascript
// above the component, after the imports
const hintTextLoading = i18n.t('Loading options...')
const hintText = i18n.t('Select dataset')

// in the component prop
hintText={props.loading ? hintTextLoading : hintText}
```
Because the string is the key, you can't have dynamic strings